### DISCUSSION TITLE
Észrevételek az eÁFA rendszerrel és a dokumentációval kapcsolatban

### DISCUSSION THREAD

**Discussion by GitARudolf** [nav-gov-hu/eVAT#49] [Ötletek - gépi interfész]
Néhány észrevételt/kérdést/ötletet/hibát gyűjtöttem össze az eÁFA rendszerrel kapcsolatban, melyeket nem feltétlen rögzítenék fel issuenak, ugyanis elképzelhető, hogy én értelmeztem félre dolgokat (pl. a dokumentáció esetén), így arra gondoltam, hogy egy discussion-t hoznék létre, ahová esetleg mások is rögzíthetik észrevételeiket.

### Dokumentációs észrevételek
A dokumentációban egy-két elírást fedeztem fel:

1. ManageDeclarationPartition operáció 11-es pont:
> A **POST /manageDeclarationUpload** végpont számára 'multipart/form-data' (RFC 2388) formátumú kérést kell küldeni, mely két üzenetből áll. Az első 'application/xml' típusú, ez az a kérés, amelyet az ManageDeclarationPartitionRequest séma element meghatároz. A másik típusa 'application/octet-stream', melyben a csatolmány található.

Itt a `/manageDeclarationUpload` végpont van megemlítve a kérési operációnak, de a `multipart/form-data` kérést a `/manageDeclarationPartition` végpontnak kell küldeni a partíciók feltöltéséhez.
Itt információként véleményem szerint szerepelhetett volna, hogy a form-data két üzenetét "body"-nak és "file"-nak kell elnevezni, mert ez az információ csak a kérés elküldését követően derül ki, a szerver válaszából, a dokumentációba nem került bele.

2. ManageDeclarationPartition operáció 12-es pont:
> A partíciók képzése a következő módon történik. **A kliens a véglegesített earData.xsd alapján képzett bevallási XML-nek kiszámítja az SHA3-512 hash értékét, majd az objektumot gzip formátumba tömöríti.** A kliens ezt követően - attól függően, hogy a tömörített objektum mérete átlépi-e a maximális partíció méretet - meghatározza, hogy a feltöltendő bevallás hány partícióból fog állni. A kliens a maximális partíció méret betartásával byte tömbökre feldarabolja a tömörített állományt és bytestreamként a multipart REST request 'application/octet-stream' üzenetében feltölti.

A kijelölt mondatban az szerepel, hogy az XML-nek ki kell számítani az SHA3-512 hash értékét, majd az objektumot (itt nem feltétlen egyértelmű, hogy a titkosított XML állományra gondolnak-e, avagy az XML dokumentumra) tömöríteni kell gzip formába.
Ahogy tapasztaltam, itt magát az XML objektumot kell betömöríteni gzip formátumban, majd ezt kell elküldeni a form-data `file` üzenetében, az SHA3-512 titkosítás, csak a request signature számításához szükséges.
Szerintem az SHA3-512 titkosítás résznek itt nem kellene szerepelnie, a dokumentációban a request signature számításnál már szerepel ez a lépés.

### Rendszerbeli észrevételek
1. Standard adókód lista
Korábban a tesztelés elején kaptunk egy standard adókód listát, de ez teljesen eltér az eÁFA rendszerben található katalógustól. Ebben még várható módosítás, vagy a katalógusban szereplővel kell dolgozni?

2. Sztornó/helyesbítő számlák a bevallásban
Mikor sztornó vagy helyesbítő számla adatát szeretnénk bevallani, akkor INVALID_TAXBASE error code-ot kapunk vissza a válasz üzenetben.

3. Részletes hibaüzenetek a bevallások elkészítésekor
A múlthét folyamán volt egy leállás az eÁFA rendszerben, a mai nap folyamán pedig amikor teszteltem, minden hibaüzenetben csak annyi szerepelt, hogy: Feldolgozás sikertelen!
Itt a leállás előtt részletesebb hibaüzenetek jelentek meg, amik egyszerűsítették a rendszer használatát.

**Comment by renced42** [comment]
Kedves @GitARudolf 

Köszönjük az észrevételeket.

Az észrevételek alapján ha jó láttam a kollégák, módosították a specit.
Az adókód lista folyamatosan változott, de a felületen a beszédes adókódok jelennek meg, nem a standard adókódok. Természetesen a háttérben ennek meg van az 1-1-es megfeleltetése.
A visszaadható hibaüzeneteket a speci tartalmazza.
